spring-cloud-netflix-core/src/test/java/org/springframework/cloud/netflix/feign/hystrix/TestInterceptor.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2013-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.netflix.feign.hystrix;
+
+import java.util.Base64;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.core.context.SecurityContextHolder;
+
+import feign.RequestInterceptor;
+import feign.RequestTemplate;
+
+/**
+ * @author Daniel Lavoie
+ */
+@Configuration
+public class TestInterceptor implements RequestInterceptor {
+	@Value("${security.user.username}")
+	private String username;
+
+	@Value("${security.user.password}")
+	private String password;
+
+	@Override
+	public void apply(RequestTemplate template) {
+		if (SecurityContextHolder.getContext().getAuthentication() != null)
+			template.header("username",
+					SecurityContextHolder.getContext().getAuthentication().getName());
+
+		template.header("Authorization", "Basic " + new String(
+				Base64.getEncoder().encode((username + ":" + password).getBytes())));
```

Comment:
I'm not sure what this is doing yet. There are 2 usernames in play here, one from the security context and one that was injected as a `@Value`. What is the header for?